packages/agent/src/actor.ts

```diff
@@ -525,35 +525,51 @@ function _createActorMethod(
       const ecid = effectiveCanisterId !== undefined ? Principal.from(effectiveCanisterId) : cid;
       const arg = IDL.encode(func.argTypes, args);
 
+      if (agent.rootKey == null)
+        throw new AgentError('Agent root key not initialized before making call');
+
       const { requestId, response, requestDetails } = await agent.call(cid, {
         methodName,
         arg,
         effectiveCanisterId: ecid,
       });
-
-      requestId;
-      response;
-      requestDetails;
-
-      if (!response.ok || response.body /* IC-1462 */) {
-        throw new UpdateCallRejectedError(cid, methodName, requestId, response);
+      let reply: ArrayBuffer | undefined;
+      let certificate: Certificate | undefined;
+      if (response.body && response.body.certificate) {
+        const cert = response.body.certificate;
+        certificate = await Certificate.create({
+          certificate: bufFromBufLike(cert),
+          rootKey: agent.rootKey,
+          canisterId: Principal.from(canisterId),
+          blsVerify,
+        });
+        const path = [new TextEncoder().encode('request_status'), requestId];
+        const status = new TextDecoder().decode(
+          lookupResultToBuffer(certificate.lookup([...path, 'status'])),
+        );
+
+        switch (status) {
+          case 'replied':
+            reply = lookupResultToBuffer(certificate.lookup([...path, 'reply']));
+            break;
+          case 'rejected':
+            throw new UpdateCallRejectedError(cid, methodName, requestId, response);
+          case 'default':
```

Comment:
Just for my understanding, is this equivalent to adding a `default:` to a `switch` stmt? In that case, we wouldn't want it to throw but go into the polling loop. 

Maybe making it more explicit would make sense? 
- received, processing -> break and poll for response 
- done or default -> throw ActorCallError

Reply:
It is not the same as a default case - I wrote this originally a few weeks ago, and can't recall why this was here. It can simply be removed and the logic below will apply